lib/browser/api/menu.js

```diff
@@ -158,23 +158,26 @@ Menu.buildFromTemplate = function (template) {
   if (!Array.isArray(template)) {
     throw new TypeError('Invalid template for Menu')
   }
-
   const menu = new Menu()
+  if (!isValidTemplate(template)) {
+    throw new TypeError('Invalid template for MenuItem: must have at least one of label, role or type')
+  }
   const filtered = removeExtraSeparators(template)
   const sorted = sortTemplate(filtered)
 
-  sorted.forEach((item) => {
-    if (typeof item !== 'object') {
-      throw new TypeError('Invalid template for MenuItem')
-    }
-    menu.append(new MenuItem(item))
-  })
+  sorted.forEach((item) => menu.append(new MenuItem(item)))
 
   return menu
 }
 
 /* Helper Functions */
 
+// validate the template against having the wrong attribute
+function isValidTemplate (template) {
+  return template.every(item =>
```

Comment:
For this wouldn't we want to throw an error signifying the fact that the template must be an array. Wouldn't this also mean keeping the logic as is, since returning false doesn't exactly tell us why  `isValidTemplate` function failed, just that it did. If we want to be more clear perhaps I could rename the function `areValidTemplateItems` since it's essentially validating the `MenuItems`. Then we could leave the `isArray` check in `Menu.buildFromTemplate` so we can throw a more descriptive error something perhaps like `Invalid template for Menu: Menu Template must be an array`.

Reply:
I don't care enough to argue about it =)